src/gui/DatabaseTabWidget.cpp

```diff
@@ -336,6 +341,22 @@ bool DatabaseTabWidget::saveDatabase(Database* db, QString filePath)
         } else {
             dbStruct.modified = true;
             updateTabName(db);
+
+            if (++dbStruct.saveAttempts > 2 && useAtomicSaves) {
+                // Saving failed 3 times, issue a warning and attempt to resolve
+                auto choice = MessageBox::question(this, tr("Disable safe saves?"),
+                                                   tr("KeePassXC has failed to save the database multiple times. "
+                                                   "This is likely caused by file sync services holding a lock on "
+                                                   "the save file.\nDisable safe saves and try again?"),
```

Comment:
@droidmonkey shouldn't we also use the `atomic saves` terminology here?

Reply:
I was thinking that, too. But I don't know how well users will understand it. Atomic in a non-nuklear context is a very domain-specific word.

Reply:
Atomic is a developer term. I wanted to dumb it down on the user interface.

Reply:
User (me) googled it: "write to tmp-file and then (atomic->) move over". (had no idea before searching for the term - sounded a little intimidating to me :) But some little hint texts, should do the trick - I guess.